src/main/java/com/mitchellbosecke/pebble/node/CacheNode.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * This file is part of Pebble.
+ *
+ * Copyright (c) 2014 by Mitchell Bösecke
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ ******************************************************************************/
+package com.mitchellbosecke.pebble.node;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.mitchellbosecke.pebble.error.PebbleException;
+import com.mitchellbosecke.pebble.extension.NodeVisitor;
+import com.mitchellbosecke.pebble.template.EvaluationContext;
+import com.mitchellbosecke.pebble.template.PebbleTemplateImpl;
+
+public class CacheNode extends AbstractRenderableNode {
+
+    private final BodyNode body;
+    private final Cache<String, String> cache;
```

Comment:
Say I want to cache some part of the header of my page, and some part of the footer. These will be two distinct `cache` tags. This will generate two different `CacheNode` instances in the tree.